test/utils.py

```diff
@@ -107,23 +108,34 @@ def get_experiment_status(status_url):
 
 def get_succeeded_trial_num(trial_jobs_url):
     trial_jobs = requests.get(trial_jobs_url).json()
-    print(trial_jobs)
     num_succeed = 0
     for trial_job in trial_jobs:
         if trial_job['status'] in ['SUCCEEDED', 'EARLY_STOPPED']:
             num_succeed += 1
     print('num_succeed:', num_succeed)
     return num_succeed
 
-def print_stderr(trial_jobs_url):
+def get_failed_trial_jobs(trial_jobs_url):
+    '''Return failed trial jobs'''
     trial_jobs = requests.get(trial_jobs_url).json()
+    failed_jobs = []
     for trial_job in trial_jobs:
-        if trial_job['status'] == 'FAILED':
-            stderr_path = trial_job['stderrPath'].split(':')[-1]
-            if sys.platform == "win32":
-                subprocess.run(['type', stderr_path], shell=True)
-            else:
-                subprocess.run(['cat', stderr_path])
+        if trial_job['status'] in ['FAILED']:
+            failed_jobs.append(trial_job)
+    return failed_jobs
+
+def print_failed_job_log(training_service, trial_jobs_url):
+    '''Print job log of FAILED trial jobs'''
+    trial_jobs = get_failed_trial_jobs(trial_jobs_url)
+    for trial_job in trial_jobs:
+        if training_service == 'local':
+            log_filename = trial_job['stderrPath'].split(':')[-1]
+        else:
+            log_filename = os.path.join(get_nni_log_dir(EXPERIMENT_URL), 'trial_' + trial_job['id'] + '.log')
```

Comment:
for the training services other than local, how can you get trial log's content?

Reply:
you enabled log collection for these tests?

Reply:
I am curious about that why in logcollection mode trial's log is stored under `experiment_id/log/` rather than `experiment_id/trials/trial_id/`?

Reply:
The logCollection for IT are enabled by this PR in order to retrieve logs.
@SparkSnail @QuanluZhang   It could be a little bit confusing if collected logs are put under `experiment_id/trials/trial_id/` since the directory is used by training service, and it seems not good to put it under  `experiment_id/log`,  maybe it's better to put it under `experiment_id/log/log_collection` ?

Reply:
I think `experiment_id/log/log_collection/` is better than `experiment_id/log/`. But I don't get why `experiment_id/trials/trail_id/` is not proper? Is logcollection also a component of training service?

Reply:
`experiment_id/log/log_collection/` provides some kind of hints that the logs are related to logCollection.
Just check the code, the remote trainning service uses `/tmp/nni/exepriments/experiment_id/trials/trial_id`  as trial dir, it is not same as this one,  so either are OK for me.

Reply:
personally, I prefer putting log under `experiment_id/trials/trial_id/`

Reply:
@QuanluZhang   Updated, changed to log directory to `experiment_id/trials/trial_id/`